data/txHandler.go

```diff
@@ -166,38 +166,44 @@ func (handler *TxHandler) backlogWorker() {
 			}
 			// enqueue the task to the verification pool.
 			handler.txVerificationPool.EnqueueBacklog(handler.ctx, handler.asyncVerifySignature, wi, nil)
+
 		case wi, ok := <-handler.postVerificationQueue:
 			if !ok {
 				return
 			}
-			if wi.verificationErr != nil {
-				// disconnect from peer.
-				logging.Base().Warnf("Received a malformed txn %v: %v", wi.unverifiedTxGroup, wi.verificationErr)
-				handler.net.Disconnect(wi.rawmsg.Sender)
-				continue
-			}
-
-			// we've processed this message, so increase the counter.
-			transactionMessagesHandled.Inc(nil)
-
-			// at this point, we've verified the transaction, so we can safely treat the transaction as a verified transaction.
-			verifiedTxGroup := wi.unverifiedTxGroup
+			handler.postprocessCheckedTxn(wi)
```

Comment:
I think that you meant to replace the upper switch statement with this call as well, right ? (postprocessCheckedTxn )

Reply:
Ah, yeah, there was some git issue.

Reply:
FYI There was also a metric that wasn't being logged correctly before. This should fix that.